nifi-manifest/nifi-runtime-manifest-core/src/main/java/org/apache/nifi/runtime/manifest/impl/StandardRuntimeManifestBuilder.java

```diff
@@ -348,6 +351,24 @@ private void populateExtensionComponent(final ExtensionManifest extensionManifes
             providedServiceApis.forEach(providedServiceApi -> providedApiTypes.add(createProvidedApiType(providedServiceApi)));
             extensionComponent.setProvidedApiImplementations(providedApiTypes);
         }
+
+        final Restricted restricted = extension.getRestricted();
+        if (restricted != null) {
+            extensionComponent.setRestricted(true);
+            extensionComponent.setRestrictedExplanation(restricted.getGeneralRestrictionExplanation());
+            if (restricted.getRestrictions() != null) {
```

Comment:
It is minor / cosmetic but what do you think about something like this. (Not sure whether functional coding style is preferred or not):
```
Optional.ofNullable(restricted.getRestrictions())
  .map(r -> r.stream().map(this::createRestriction).collect(Collectors.toSet()))
  .ifPresent(restrictions -> extensionComponent.setExplicitRestrictions(restrictions));
```

Reply:
If everything else in the class was already in this style, then I'd definitely be consistent and do something like that. For now I'd leave it as is.